### PR TITLE
fix(memory): increase reserved mem to 30%

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

The `reserved_memory` is designed for the difference between actual usage (`jemalloc_allocated`) and process usage (`process_resident_memory`).

https://github.com/risingwavelabs/risingwave/blob/b640bc99952d4541a6f43213f57883f2798d76b6/src/compute/src/memory/config.rs#L40-L42

As being observed in #13610,

> - jemalloc_allocated_bytes: 8.59 GB
> - jemalloc_active_bytes: 9.93 GB
> - Node memory (RSS): 12.3 GB

The ratio is `8.59/12.3 = 0.698 ≈ 0.7`.

Passed longevity test: https://buildkite.com/risingwave-test/longevity-test/builds/818 (Previously this would OOM for 2 times)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/compute/src/memory/config.rs [hunk 1]
```diff
@@ -157,14 +157,14 @@ mod tests {
     #[test]
     fn test_reserve_memory_bytes() {
         // at least 512 MB
```
Comment (KeXiangWang) [inline src/compute/src/memory/config.rs:159]: Should be 1GB now.

**Review by KeXiangWang — APPROVED** [review]
LGTM

File: src/compute/src/memory/config.rs (hunk 1 above)
Comment (fuyufjh) [reply src/compute/src/memory/config.rs:159]: Any reference?

Comment (KeXiangWang) [reply src/compute/src/memory/config.rs:159]: Oh, sorry. 512MB Is correct. I misread the corresponding codes.

**Comment by wcy-fdu** [comment]
Previously I also made similar adjustments, hoping to pass the longevity test.
My change is:
```
const SYSTEM_RESERVED_MEMORY_PROPORTION: f64 = 0.35;
const STORAGE_MEMORY_PROPORTION: f64 = 0.15;
```

and the longevity test passed **without limit rdkakfa fetch_queue_size** 
https://buildkite.com/risingwave-test/longevity-test/builds/787

**Review by wcy-fdu — APPROVED** [review]

**Review by hzxa21 — APPROVED** [review]
LGTM

**Comment by yuhao-su** [comment]
Should we consider using `stats.mapped` or `stats.resident` instead of allocated in this case?

**Comment by fuyufjh** [comment]
> Should we consider using `stats.mapped` or `stats.resident` instead of allocated in this case?

`stats.mapped` or `stats.resident` is closer to the actual memory usage (rss), but similar with rss, it doesn't reflect the actual bytes allocated by the program. I would tend to use system rss if we really consider `stats.allocated` to be too inaccurate.

PS. At the beginning, I planned to replace the `jemalloc_allocated` to `system rss`, but I would like to try this "conservative" solution first.

**Pull request merged**